### ISSUE TITLE
Add ability to click on a cluster in dim red & pull up info/markers

### ISSUE THREAD

**Issue by sjessa** [fungenomics/braindex#8]

**Comment by bhavyaac** [comment]
Instead of using Plotly for an interactive plot (could be too slow / overkill), store the cluster centres in a new data frame and use nearPoints() using the centres, similar to dendrogram hover functionality. This would only be helpful in the annotation by region, since the others (by timepoints, by sample) have overlapping clusters.

**Comment by bhavyaac** [comment]
As of PR #12, cluster name is visible on hover in the dim. red. plot in Single cell expression by region (tSNE and UMAP reductions only).

**Comment by bhavyaac** [comment]
The hover tooltip is small so that it doesn't cover up the plot. If we are adding cluster information and/or markers to it, perhaps this information can be available in a separate, larger tooltip. Depending on its size, we could place it to the right of the plots or below the side bar, and activate it by click instead of hover to differentiate it from the cluster name. (Not sure if we can have separate tooltips activated by hover vs. by click.)

**Comment by sjessa** [comment]
Another idea, maybe too complicated: what if we have a separate tab containing a table with _all_ cluster markers, and it's a linked data table, so that if one clicks on a cluster in the dim red space, it will automatically filter the datatable to the markers for that cluster?

See here for general idea: https://community.rstudio.com/t/link-dt-datatable-to-a-ggplot/48458

If this is too complicated, we can just have the searchable/filter-able table in another tab, no fancy linking to ggplot/hover needed :)

**Comment by bhavyaac** [comment]
I like this idea the most:
> e.g. in the "Expression table" tab, right now, clicking on rows has no effect, but maybe the marker table could be below, and it could be automatically filtered based on the rows clicked in the first table?

I think it would be most intuitive for the markers to show up in the same tab as the clicking action, and the dimensionality reduction tabs are already pretty full.

**Comment by sjessa** [comment]
Agreed! Plus it doesn't make sense that clicking on specifically the dimensionality reduction would filter the table, since there are so many different places in the app to click...

Having two linked tables in that tab sounds good!

**Comment by bhavyaac** [comment]
### Required documentation

- In the table containing 100 signature genes, I will describe statistics in the following way:
> Pct_1: "detection rate in cluster"
> Pct_2: "detection rate out of cluster"
> Pct_1 - Pct_2: "difference in detection" (aka "specificity") 

I may need to use abbreviated versions of these in the table header itself, and these abbreviations should be explained in the tab

- I will have to document the binarized expression used for calculating a detection rate, as well as the way that the 100 signature genes are picked in the app wiki
- I will replace the current tab name "Expression table" with "Cluster information and markers"

**Comment by bhavyaac** [comment]
I've created the linked table! 
![marker table](https://user-images.githubusercontent.com/26724860/133176304-f96350d5-0218-4220-b6bb-83640c329379.gif)

Let me know if you have any feedback @sjessa, especially on the documentation:
![image](https://user-images.githubusercontent.com/26724860/133176390-7e72d804-ba2b-482c-ad38-db7c767d7feb.png)

**Comment by bhavyaac** [comment]
Update: I've included the gene biotype and added some conditional formatting to the table!

<img width="875" alt="Screen Shot 2021-09-14 at 11 43 54 AM" src="https://user-images.githubusercontent.com/26724860/133290185-72049720-4e7e-4cbd-b8c0-dd252983a202.png">

**Comment by bhavyaac** [comment]
I realized that the user may want to check cluster markers without having to enter any genes of interest in the sidebar. If this is a common issue, I can consider displaying the expression table without the genes of interest before any genes are added as input, so it can still be used for the marker table. (I would probably have to change the table title from `Expression table` to something else then!)

**Comment by sjessa** [comment]
The implementation looks good to me! I like the conditional formatting too - the only thing I would suggest is to use it more sparingly so the most important stats pop out; I would only colour the `Specificity` column, I think!

> I realized that the user may want to check cluster markers without having to enter any genes of interest in the sidebar. If this is a common issue, I can consider displaying the expression table without the genes of interest before any genes are added as input

It would definitely useful to be able to view the table w/out any genes entered. I would think of this table more as cluster-level info, which will happen to display expression if any genes are entered. For the tab title, maybe we can just use "cluster information" to capture that.
